cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -86,15 +143,29 @@ func LockWithHint(snapName string, hint Hint) error {
 	}
 	defer flock.Close()
 
+	// The following order of execution is important to avoid race conditions.
+	// Take the lock
 	if err := flock.Lock(); err != nil {
 		return err
 	}
+	// Write inhibit info
+	buf, err := json.Marshal(info)
+	if err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(InhibitInfoFile(snapName, hint), buf, 0644); err != nil {
+		return err
+	}
+	// Write hint
 	f := flock.File()
 	if err := f.Truncate(0); err != nil {
 		return err
 	}
-	_, err = f.WriteString(string(hint))
-	return err
+	if _, err = f.WriteString(string(hint)); err != nil {
+		return err
+	}
```

Comment:
I think we should do a `if err := flock.Sync(); err != nil {` here, because:
a) we do not check for flock.Close() errors in the defer so the most common case of some i/o error on close would go undetected. this the fsync() would fix this in most (all?) cases
b) close() itself does not gurantee that things are written to disk so just for reliability it's probably a good idea to fsync() here

Reply:
totally agree, thanks for pointing it out.